components/match2/commons/starcraft_starcraft2/match_group_input_starcraft_ffa.lua

```diff
@@ -314,79 +327,71 @@ end
+		OppNumber = opponentIndex
+
+		local bg = StarcraftFfaInput._bgClean(opponent.bg)
+		opponent.bg = nil
+		local advances = opponent.advance or ''
+		if advances == '' then
+			advances = opponent.win or ''
 			if advances == '' then
-				advances = match['opponent' .. opponentIndex].win or ''
-				if advances == '' then
-					advances = match['opponent' .. opponentIndex].advances or ''
-				end
-			end
-			advances = advances ~= 'false' and advances ~= '' and advances ~= '0'
-
-			--opponent processing (first part)
-			--sort out extradata
-			match['opponent' .. opponentIndex].extradata = {
-				advantage = match['opponent' .. opponentIndex].advantage,
-				score2 = match['opponent' .. opponentIndex].score2,
-				isarchon = match['opponent' .. opponentIndex].isarchon,
-				advances = advances,
-				noscore = noscore,
-				bg = bg
-			}
-
-			--set initial opponent sumscore
-			match['opponent' .. opponentIndex].sumscore =
-				tonumber(match['opponent' .. opponentIndex].extradata.advantage or '') or ''
-
-			local temp_place = match['opponent' .. opponentIndex].placement
-			--process input depending on type
-			if match['opponent' .. opponentIndex]['type'] == 'solo' then
-				match['opponent' .. opponentIndex] =
-					StarcraftMatchGroupInput.ProcessSoloOpponentInput(match['opponent' .. opponentIndex])
-			elseif match['opponent' .. opponentIndex]['type'] == 'duo' then
-				match['opponent' .. opponentIndex] =
-					StarcraftMatchGroupInput.ProcessDuoOpponentInput(match['opponent' .. opponentIndex])
-			elseif match['opponent' .. opponentIndex]['type'] == 'trio' then
-				match['opponent' .. opponentIndex] =
-					StarcraftMatchGroupInput.ProcessOpponentInput(match['opponent' .. opponentIndex], 3)
-			elseif match['opponent' .. opponentIndex]['type'] == 'quad' then
-				match['opponent' .. opponentIndex] =
-					StarcraftMatchGroupInput.ProcessOpponentInput(match['opponent' .. opponentIndex], 4)
-			elseif match['opponent' .. opponentIndex]['type'] == 'team' then
-				match['opponent' .. opponentIndex] =
-					StarcraftMatchGroupInput.ProcessTeamOpponentInput(match['opponent' .. opponentIndex], match.date)
-			elseif match['opponent' .. opponentIndex]['type'] == 'literal' then
-				match['opponent' .. opponentIndex] =
-					StarcraftMatchGroupInput.ProcessLiteralOpponentInput(match['opponent' .. opponentIndex])
-			else
-				error('Unsupported Opponent Type')
+				advances = opponent.advances or ''
 			end
+		end
+		advances = advances ~= 'false' and advances ~= '' and advances ~= '0'
+
+		--opponent processing (first part)
+		--sort out extradata
+		opponent.extradata = {
+			advances = advances,
+			advantage = opponent.advantage,
+			bg = bg,
+			isarchon = opponent.isarchon,
+			noscore = noscore,
+			score2 = opponent.score2,
+		}
+
+		--set initial opponent sumscore
+		opponent.sumscore =
+			tonumber(opponent.extradata.advantage or '') or ''
+
+		local inputPlace = opponent.placement
```

Comment:
Why is this needed? (and Line 380)

Reply:
yes it is needed
the default input processing (functions in the non ffa module) sets placements 1/2 which are just wrong for ffa
will add a comment